OneCommunityGlobal/HighestGoodNetworkApp
# Sola Implement Summary Management Features

# Description
This PR completes the work started on PR 838 (Summary Management). It implements all the features required in the front end to allow for a summary receiver (manager or mentor) to review the weekly summaries of everyone on their summary team in a simple and quick way. It also implements a way to see which summary teams a volunteer belongs to under "profile". Finally, it implements a way for an admin to create, modify or delete summary teams and populate the fields with volunteers and managers/mentors.

## Related PRS (if any):
This frontend PR is related to the #358 backend PR and continues the work on #838 on the front end.
…

## Main changes explained:
The PR implements CRUD operations for Summary Management functions.
…

## How to test:
1. check into the current branch on the front end and PR 358 on the backend
2. do pm install and nom run start:local to run this PR locally
3. watch the attached demo video to see what features are included and should be tested.
4. log in as Admin and check that "Summary Management" shows up under "other links" in the nav bar. Also check that you can add, remove and modify summary teams. Create teams and populate the team members and summary receiver fields.
5. Go under "Profile" in any role and see that the user can see the summary teams to which they belong under "Summary
Teams" tab.
6. Create an account with Manager or Mentor role and check that you can see Summary Management in the nav bar. Check that you can view reports of all teams in which the manager and mentor user is a summary receiver.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/a6ccfb79-114b-4693-9190-a21d15196a87

## Note:
Though, the PR has more changes than usual to test. Some of the tests listed in the video have been tested in previous PRs e.g., PR #838.
